platforms/core-configuration/model-core/src/main/java/org/gradle/model/internal/asm/AsmClassGeneratorUtils.java

```diff
@@ -39,14 +40,20 @@ public static String signature(Constructor<?> constructor) {
      */
     public static String signature(Constructor<?> constructor, boolean addNameParameter) {
```

Comment:
Looks good!

This method and the variant that takes a Method strike me as something that would be better unit-tested than integration-tested (much easier to exercise different scenarios, easier to understand expectations too).

Also, it seems the other variant of this method (which takes only the constructor) is not used anywhere.

Reply:
Thanks, @abstratt!